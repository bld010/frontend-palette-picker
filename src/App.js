import React, { Component } from 'react';
import ColorDisplay from './ColorDisplay';
import './App.scss';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currentPalette: null
    }
  }

  render = () => {

    return(
      <main className="App">
        <h1>Palette Picker</h1>
        <ColorDisplay palette={this.state.currentPalette} />
      </main>
    )
  }
}

export default App;
