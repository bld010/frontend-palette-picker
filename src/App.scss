.App {
  text-align: center;

  & section {
    background-color: pink;
    display: flex;
    flex-direction: row;
  }

  & .Folders {
    background-color: green;
    width: 400px;
  }
}
