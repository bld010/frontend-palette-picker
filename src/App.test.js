import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('App', () => {

  let wrapper;

  let mockPalette1 = {
    folder_id: 1,
    id: 22,
    colors: [],
    name: "Mock Palette 1"
  }

  let mockPalette2 = {
    folder_id: 1,
    id: 23,
    colors: [],
    name: "Mock Palette 2"
  }

  let mockPalette3 = {
    folder_id: 2,
    id: 24,
    colors: [],
    name: "Mock Palette 3"
  }

  let mockPalette4 = {
    folder_id: 2,
    id: 24,
    colors: [],
    name: "Mock Palette 3"
  }

  let mockFolders = [
    { name: "Folder 1", 
      id: 1,
      palettes: [
        mockPalette1,
        mockPalette2
      ], 
    }, 
    { name: "Folder 2", 
      id: 2,
      palettes: [
        mockPalette3,
        mockPalette4
      ], 
    }, 
    
  ]

  beforeEach(() => {
    wrapper = shallow(<App />)
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('displayFolderPalettes', () => {
    it('should set state with correct folder based on id', async () => {
      
      await wrapper.instance().setState({folders: mockFolders})

      wrapper.instance().displayFolderPalettes(1)

      let expected = { 
        name: "Folder 1", 
        id: 1,
        palettes: [
          mockPalette1,
          mockPalette2
        ], 
      }
      
      expect(wrapper.state().currentFolder).toEqual(expected);
      
    })
  })

  describe('setCurrentPalette', () => {
    it('should set state with the palette selected', () => {

    })
  })

  describe('reAssignData', () => {
    it('should fire getPalettes', () => {

    })

    it('should fire cleanPalettes with the fetched palettes', () => {

    })

    it('should fire getFolders', () => {

    })

    it('should fire cleanFolders with the fetched folders', () => {

    })

    it('should fire cleanFolders with the cleaned palettes and folders', () => {

    })

    it('should set state with the cleaned data', () => {

    })
  })

  describe('deleteFolder', () => {
    it('should fire deleteFolder with the correct id', () => {

    })

    it('should fire reAssignData', () => {

    })

    it('should set the currentFolder property to null', () => {

    })
  })

  describe('deletePalette', () => {
    it('should fire deletePalette with the correct id', () => {

    })

    it('should fire reAssignData', () => {

    })

    it('should update the palettes in currentFolder in state with deleted palette removed', () => {

    })
  })

  describe('componentDidMount', () => {
    it('should fire reAssignData', () => {

    })

    it('should set state with an error message if a fetch call fails', () => {

    })
  })

})

