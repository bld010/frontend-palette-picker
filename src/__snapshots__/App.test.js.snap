// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<main
  className="App"
>
  <h1>
    Palette Picker
  </h1>
  <ColorDisplay
    displayFolderPalettes={[Function]}
    folders={Array []}
    palette={null}
    reAssignData={[Function]}
  />
  <section>
    <div
      className="folders"
    >
      <h3>
        Folders
      </h3>
      <Folders
        deleteFolder={[Function]}
        displayFolderPalettes={[Function]}
        folders={Array []}
      />
    </div>
    <div
      className="palettes"
    >
      <h3>
        Palettes 
      </h3>
      <Palettes
        setCurrentPalette={[Function]}
      />
    </div>
  </section>
</main>
`;
